include/odp/api/spec/traffic_mngr.h

```diff
@@ -331,6 +331,26 @@ typedef struct {
 	/** The per_level array specifies the TM system capabilities that
 	 * can vary based upon the tm_node level. */
 	odp_tm_level_capabilities_t per_level[ODP_TM_MAX_LEVELS];
+
+	/** node_priority indicates that this TM system supports
+	 * only having single priority associated to fan-in nodes at any
+	 * level. When TRUE, Priority is fixed for a fanin node i.e all packets
+	 * fed by a particular fanin node to a tm node will have a fixed priority.
+	 * This is equivalent to the packet priority being overloaded by
+	 * fan-in node priority. When FALSE, every fan-in node can feed packets
+	 * to parent tm node with priorities ranging from
+	 * 0..ODP_TM_MAX_PRIORITIES - 1.
+	 */
+	odp_bool_t node_priority_override;
```

Comment:
This capability now makes the "node priority override" and the old behaviour mutually exclusive, but we can have implementations, like linux-gen, which can easily support both. I think it would be better to have separate capabilities for both types of priority handling (so an implementation could set them both) and have a global or per-node config setting for selecting which one to use.

Reply:
Ack. I'll add it as mode in odp_tm_capabilities_t and odp_tm_requirements_t so that per-TM, user can mention the mode based on capability while doing odp_tm_create().

Per-node config will be complicated to visualize/design the topology I think.

Reply:
Sounds ok to me.

Reply:
As discussed, node_priority_override sounds a bit weird (we are not overriding the priority of a node). Maybe packet_priority_override, priority_override or priority_rewrite or something like that would be better?